core/src/main/scala/kafka/producer/async/DefaultEventHandler.scala

```diff
@@ -129,9 +129,22 @@ class DefaultEventHandler[K,V](config: ProducerConfig,
     events.foreach{e =>
       try {
         if(e.hasKey)
-          serializedMessages += new KeyedMessage[K,Message](topic = e.topic, key = e.key, partKey = e.partKey, message = new Message(key = keyEncoder.toBytes(e.key), bytes = encoder.toBytes(e.message)))
+          serializedMessages += new KeyedMessage[K,Message](
+            topic = e.topic,
+            key = e.key,
+            partKey = e.partKey,
+            message = new Message(key = keyEncoder.toBytes(e.key),
+                                  bytes = encoder.toBytes(e.message),
+                                  timestamp = Message.NoTimestamp,
+                                  magicValue = Message.MagicValue_V0))
```

Comment:
Hi Jun, do we also want the change in old producer request and old producer? It seems it will be deprecated pretty soon. So in the current patch I did not even change the scala ProducerRequest format, but I probably should just leave the scala producer version to 1.

Besides that, I realized that we have quite a few tools still using old consumer. I will update them.